fix: Don't override rules from user's config

melusc opened the pull request:
If the user disables any of `n/no-unsupported-features/*` or otherwise overrides it in their config it will now keep that setting.

Fixes #598 and #760
Closes #672

melusc commented:
I haven't looked into it, but I would imagine <https://github.com/xojs/xo/blob/112e6579181d4080897c4ff1e2d1ab45da0180a3/lib/options-manager.js#L364> and similar lines should also use `??=` instead to respect the user's config. I can make a PR or include that in this PR if desired

sindresorhus commented:
> I can make a PR or include that in this PR if desired

Would be great if you could include it in this PR :)

melusc commented:
I have updated them all to use nullish coallescing assignments. I haven't added any more tests. I think if something changes that would undo this the test I already added should catch anything.
